### PR TITLE
Fix the TabTests, April 2022 edition

### LINKED ISSUE #12158: TerminalAppLocalTests::TabTests are failing
### Windows Terminal version

Commit c3fd26a393bd00664a3527f3c973e2cec2344e06

### Windows build number

10.0.19041.1415

### Other Software

_No response_

### Steps to reproduce

1. Checkout the latest code.
2. Build. 
3. Run the `TerminalAppLocalTests` unit tests.

### Expected Behavior

Tests should pass.

### Actual Behavior

Two tests are failing:

    TerminalAppLocalTests::TabTests::TestPreviewCommitScheme [Failed]
    TerminalAppLocalTests::TabTests::TestPreviewDismissScheme [Failed]

As far as I can tell, these first stopped working in commit c3fd26a393bd00664a3527f3c973e2cec2344e06.

### PR DESCRIPTION
**Pull request by zadjii-msft**
In classic fashion, we never run the LocalTests locally before committing, so stuff breaks from time to time.

This time, the main trick was that the tests had a pretty hardcore dependency on the inner workings of `_PreviewActionHandler`, and when that changed, they broke.

Also, there was a weird crash I saw when I had the default terminal set to the Dev build version. That crash would let the test contents pass, but ultimately fail when TAEF tore down the conhost. Unsetting that fixed the crash 🤷 

Closes #12158

### CODE REVIEW

**Review by DHowett — APPROVED** [review]

**Review by raghavthind2005 — APPROVED** [review]

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -3380,6 +3380,12 @@ namespace winrt::TerminalApp::implementation
     // - Displays a info popup guiding the user into setting their default terminal.
     void TerminalPage::ShowSetAsDefaultInfoBar() const
     {
+        if (::winrt::Windows::UI::Xaml::Application::Current().try_as<::winrt::TerminalApp::App>() == nullptr)
+        {
+            // Just ignore this in the tests (where the Application::Current()
+            // is not a TerminalApp::App)
+            return;
+        }
```
Comment (lhecker) [inline src/cascadia/TerminalApp/TerminalPage.cpp:3388]: That feels like a rather... unique solution. 😄
Is there some other way to achieve this?

File: src/cascadia/TerminalApp/ActionPreviewHandlers.cpp [hunk 2]
```diff
@@ -140,6 +140,33 @@ namespace winrt::TerminalApp::implementation
         });
     }
 
+    void TerminalPage::_PreviewAction(const Settings::Model::ActionAndArgs& args)
```
Comment (lhecker) [inline src/cascadia/TerminalApp/ActionPreviewHandlers.cpp:143]: Does it make sense to call this function from `_PreviewActionHandler` to deduplicate the code?

File: src/cascadia/TerminalApp/ActionPreviewHandlers.cpp [hunk 3]
```diff
@@ -140,6 +140,33 @@ namespace winrt::TerminalApp::implementation
...
+        case ShortcutAction::SetColorScheme:
+        {
+            _PreviewColorScheme(args.Args().try_as<SetColorSchemeArgs>());
+            break;
+        }
+        case ShortcutAction::AdjustOpacity:
+        {
+            _PreviewAdjustOpacity(args.Args().try_as<AdjustOpacityArgs>());
+            break;
+        }
```
Comment (lhecker) [inline src/cascadia/TerminalApp/ActionPreviewHandlers.cpp:156]: These brackets aren't needed.

File: src/cascadia/LocalTests_TerminalApp/TabTests.cpp [hunk 4]
```diff
@@ -1296,6 +1300,8 @@ namespace TerminalAppLocalTests
     {
         Log::Comment(L"Preview a color scheme. Make sure it's applied, then committed accordingly");
 
+        // DebugBreak();
+
```
Comment (carlos-zamora) [inline src/cascadia/LocalTests_TerminalApp/TabTests.cpp:1304]: You can remove this

**Review by carlos-zamora — COMMENTED** [review]
I'll let Leonard handle giving you the second sign off. I'm really just here to ask if we should try and run the LocalTests as a part of our bug bashes? We _should_ do it on every commit (or at least the relevant ones), but maybe a half-step like this would help? Plus, we want these to go into the helix tests right?

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp:3388]: Meh, there's actually a decent amount of precedent for doing this elsewhere in the app. It's not super semantically clean, but it does _work_ quite well.

File: src/cascadia/TerminalApp/ActionPreviewHandlers.cpp (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/ActionPreviewHandlers.cpp:143]: derp that was the whole point 🤦

**Comment by zadjii-msft** [comment]
> Plus, we want these to go into the helix tests right?

They actually already are! we still just don't get emails from Helix, or something?

**Pull request merged**